dea/apps/index_from_tar.py

```diff
@@ -64,7 +64,7 @@ def cli(input_fname,
     if ignore_lineage:
         auto_add_lineage = False
 
-    def mk_s3_uri(name):
+    def mk_uri(protocol, name):
```

Comment:
`protocol` is already captured don't pass it in

Also move `protocol` normalisation if statement out of this function, no point doing it on every call.